src/SimulationRunner.cc

```diff
@@ -1178,6 +1198,22 @@ bool SimulationRunner::OnWorldControlState(const msgs::WorldControlState &_req,
   return true;
 }
 
+/////////////////////////////////////////////////
+void SimulationRunner::ProcessNewWorldControlState()
+{
+  std::lock_guard<std::mutex> lock(this->msgBufferMutex);
+  // update the server ECM if the request contains SerializedState information
+  if (this->newWorldControlState && this->newWorldControlState->has_state())
+  {
+    this->entityCompMgr.SetState(this->newWorldControlState->state());
```

Comment:
It looks like this code path isn't tested. It would be nice to make sure that the message has the intended results on the ECM.

Reply:
Captured in #1232